### PR TITLE
Added deployment flag to the add command

### PR DESCRIPTION
**Pull request by josecordaz**
Fix: #157 
Fix: #181

### CODE REVIEW

**Review by jrryjcksn — APPROVED** [review]
LGTM except I think a unit test is failing because the params.DeploymentType isn't given a value and the defaulting in the command doesn't happen. Once that's fixed, :+1:

File: pkg/cmdimpl/add.go [hunk 1]
```diff
@@ -337,10 +346,13 @@ func Add(args []string, allParams AddParamSet) {
 	source := generateSourceManifest()
 
 	var appManifests []byte
-	if params.DeploymentType == DeployTypeHelm {
+	switch params.DeploymentType {
+	case string(DeployTypeHelm):
```
[the remark this answers is not in the document]
Comment (josecordaz) [reply pkg/cmdimpl/add.go:350]: correct, i have changes in another branch where I created a DeploymentType type and this wrapping is needed. Forgot to `remove string(...)` on this code 🙂 .

**Comment by josecordaz** [comment]
> LGTM except I think a unit test is failing because the params.DeploymentType isn't given a value and the defaulting in the command doesn't happen. Once that's fixed, 👍

@jrryjcksn This would be the place right? https://github.com/weaveworks/weave-gitops/blob/deployment-flag/test/acceptance/test/core_operations_test.go#L116

**Comment by jrryjcksn** [comment]
Yes, sorry I didn't see this sooner!

**Pull request merged**